doc/foreshadow_sequence_main.uml

```diff
@@ -0,0 +1,68 @@
+@startuml
+
+skinparam Shadowing false
+
+participant User
+
+User -> Foreshadow: ~__init__()
+
+note over Foreshadow, DataPreparer
+    Foreshadow and DataPreparer
+    are wrappers for an sklearn
+    Pipeline. Fit and transform
+    calls originate from sklearn
+    logic and not the objects
+    themselves.
+end note
+
+note over Foreshadow
+    shadow.steps = Pipeline([
+        ('t', DataPreparer()),
+        ('m', LogisticRegression()),
+    ])
+end note
+
+note over DataPreparer
+    ci = ColumnInfoSharer()
+    dp.steps = Pipeline([
+        ('c', FeatureCleaner(share=ci)),
+        ('i', IntentResolver(share=ci)),
+        ('e', FeatureEngineerer(share=ci)),
+        ('p', FeaturePreprocessor(share=ci)),
+        ('r', FeatureReducer(share=ci)),
+    ])
+end note
+
+User -> Foreshadow: fit(X, y)
+Foreshadow -> DataPreparer: fit(X, y)
+
+DataPreparer -> ColumnInfoSharer: ~__init__()
+
+DataPreparer -> FeatureCleaner: fit_transform(X, y)
+FeatureCleaner -> ColumnInfoSharer: set_info(column, 'tags', values)
+FeatureCleaner --> DataPreparer: X'
```

Comment:
I think we should either do X1, X2, X3 etc. or just keep it all as X. I tend to prefer the latter.

Reply:
makes sense.